src/main/java/school/lemon/changerequest/java/io/hw1/IFileManagerImp.java

```diff
@@ -0,0 +1,625 @@
+    public void createFile(String name) throws FileAlreadyExistException, FileManagerException {
+        File file = new File(name);
+
+        try {
+            if (file.isDirectory()) {
+                throw new IOException();
+            }
+            boolean fcnf = file.createNewFile();
+            if (fcnf) {
+                System.out.println("File has been created successfully");
+            } else {
+                throw new FileAlreadyExistException(file);
+            }
+        } catch (FileAlreadyExistException n) {
+            System.out.println(n.getMessage());
+        } catch (FileManagerException n) {
+            System.out.println(n.getMessage());
+        } catch (IOException e) {
+            System.out.println("Not File");
+        }
+    }
+
+
+    @Override
+    public void createDirectory(String name) throws DirectoryAlreadyExistException, FileManagerException {
+        File file = new File(name);
+        try {
+            if (file.isDirectory()) {
+                throw new IOException();
+            }
+            boolean fcnf = file.mkdir();
+            if (fcnf) {
+                System.out.println("Directory has been created successfully");
+            } else {
+                throw new DirectoryAlreadyExistException( file);
+            }
+        } catch (DirectoryAlreadyExistException n) {
+            System.out.println(n.getMessage());
+        } catch (FileManagerException n) {
+            System.out.println(n.getMessage());
+        } catch (IOException e) {
+            System.out.println("Not Directory");
+        }
+
+    }
+
+    @Override
+    public void changeDirectory(String name) throws IllegalDirectoryException {
+
+        try {
+            Paths.get(name);
+            setPath(name);
+            System.out.println("Changed Directory");
+        } catch (IllegalDirectoryException ide) {
+            System.out.println(ide.getMessage());
+        }
+    }
+
+    @Override
+    public boolean delete(String name) throws FileNotFoundException {
+        try
+        {
+            File file = new File(name);
+            if(file.exists())
+            {
+                file.delete();
+                return true;
+            }
+            else
+            {
+                throw new FileNotFoundException();
+            }
+        }
+        catch (FileNotFoundException fnfe)
+        {
+            System.out.println(fnfe.getMessage());
+        }
+        return false;
+    }
+
+    /**
+     * Delete file or directory with a given name.
+     * If force == true delete directory even if it is not empty.
+     *
+     * @param name  file or directory name to delete
+     * @param force should directory be removed even if it is not empty?
+     * @return true if file or directory was deleted
+     * @throws FileNotFoundException if file or directory could not be found
+     */
+    @Override
+    public boolean delete(String name, boolean force) throws FileNotFoundException {
```

Comment:
You shouldn't check `file.length`. `file.length` will give you only amount of file bytes.
You need to check whether file is a directory and if it is - use some recursive method to delete all files in given directory if `force==true`.